### ISSUE TITLE
Single Shard Transformer Benchmark 7x Slower with neuronx-cc 2.7

### ISSUE THREAD

**Issue by xanderdunn** [aws-neuron/aws-neuron-sdk#692]
We have a benchmark on a Transformer with hyper-parameters `(seq, n_layers, d_model, n_heads) = (2048, 1, 4608, 72)`.
XLA HLO graph: [rust_hlo_transformer_xla_benchmark.pb.zip](https://github.com/aws-neuron/aws-neuron-sdk/files/11779821/rust_hlo_transformer_xla_benchmark.pb.zip)

After upgrading to NRT SDK 2.11, we find that it benchmarks much slower.

Compile with:
```
neuronx-cc compile /tmp/rust_hlo_transformer_xla_benchmark.pb --framework XLA --target trn1 --model-type transformer --output /tmp/transformer_xla_benchmark_15811514111177462288.neff
```
It produces this compiled model: [transformer_xla_benchmark_2.7.neff.zip](https://github.com/aws-neuron/aws-neuron-sdk/files/11779888/transformer_xla_benchmark_2.7.neff.zip)

Benchmark with:
```
neuron-bench infer --fixed-instance-count 1 --enable-only-throughput -n 10 --verbose 4 /tmp/transformer_xla_benchmark_15811514111177462288.neff
```
And we see:
```
transformer_xla_benchmark_15811514111177462288
+---+----+---------+---------+---------+-------+--------+--------+---------+---------+--------+---------+---------+-------+
  B   NC   NC USED   WEIGHTS   MODE      INF/S   IRES/S   L(1)     L(50)     L(99)     NCL(1)   NCL(50)   NCL(99)   %USER
  1   1    1         dynamic   LIBMODE   1.13    1.13     925943   1770772   1772439   884248   885368    885972    0.06
+---+----+---------+---------+---------+-------+--------+--------+---------+---------+--------+---------+---------+-------+
```

This is on a trn1.32xlarge with all latest neuron versions:
```
$ dpkg -l | grep -i neuron
ii  aws-neuronx-collectives                2.14.9.0-f3942f597                amd64        neuron_ccom built using CMake
ii  aws-neuronx-dkms                       2.10.11.0                         amd64        aws-neuronx driver in DKMS format.
ii  aws-neuronx-oci-hook                   2.2.9.0                           amd64        neuron_oci_hook built using CMake
ii  aws-neuronx-runtime-lib                2.14.8.0-edaf1b171                amd64        neuron_runtime built using CMake
ii  aws-neuronx-tools                      2.11.10.0                         amd64        Neuron profile and debug tools
$ neuronx-cc --version
NeuronX Compiler version 2.7.0.40+f7c6cf2a3

Running on AMI ami-01257e71ecb2f431c
Running in region usw2-az4
Running on python 3.8.10

HWM version 2.7.0.3-0092b9d34
NumPy version 1.21.6
```

----

So, I rolled back the neuronx-cc version to previous:
```
pip3 uninstall neuronx-cc neuronx-hwm
pip3 install neuronx-cc==2.6.0.19
```

So we have the new runtime and the old compiler:
```
$ dpkg -l | grep -i neuron
ii  aws-neuronx-collectives                2.14.9.0-f3942f597                amd64        neuron_ccom built using CMake
ii  aws-neuronx-dkms                       2.10.11.0                         amd64        aws-neuronx driver in DKMS format.
ii  aws-neuronx-oci-hook                   2.2.9.0                           amd64        neuron_oci_hook built using CMake
ii  aws-neuronx-runtime-lib                2.14.8.0-edaf1b171                amd64        neuron_runtime built using CMake
ii  aws-neuronx-tools                      2.11.10.0                         amd64        Neuron profile and debug tools

$ neuronx-cc --version
NeuronX Compiler version 2.6.0.19+3d819e565

Python version 3.8.10
HWM version 2.6.0.0-826e77395
NEFF version Dynamic
TVM not available
NumPy version 1.21.6
MXNet not available
```

It produces this compiled model: [transformer_xla_benchmark_2.6.neff.zip](https://github.com/aws-neuron/aws-neuron-sdk/files/11779893/transformer_xla_benchmark_2.6.neff.zip)

And get the result:
```
transformer_xla_benchmark_15811514111177462288
+---+----+---------+---------+---------+-------+--------+--------+--------+--------+--------+---------+---------+-------+
  B   NC   NC USED   WEIGHTS   MODE      INF/S   IRES/S   L(1)     L(50)    L(99)    NCL(1)   NCL(50)   NCL(99)   %USER
  1   1    1         dynamic   LIBMODE   8.14    8.14     161883   241622   242787   120198   120813    120923    0.17
+---+----+---------+---------+---------+-------+--------+--------+--------+--------+--------+---------+---------+-------+
```
Compare 1.13 INF/s vs. 8.14 INF/s. On the old version of neuronx-cc, **this is 7x faster!**

This indicates that it's specifically the new version of `neuronx-cc` that's compiling this to something much less performant.

**Comment by xanderdunn** [comment]
This is fixed in neuronx-cc 2.8 beta:
```
$ neuronx-cc --version
NeuronX Compiler version 2.8.0.13+a76a82727

Running on AMI ami-060e1cfd2b2588d6b
Running in region usw2-az4
Running on python 3.8.10

HWM version 2.8.0.3-2b7c6da39
NumPy version 1.21.6
```

In fact, we see a 10-15% improvement over previous best!
(seq, n_layers, d_model, n_heads): previous best -> new best (lower better)
(2048, 1, 4608, 72): 0.13111s -> 0.11235s
(2048, 1, 5120, 80): 0.14139s -> 0.12920s

Huge thanks for the significant performance improvement! 🥇

**Comment by aws-donkrets** [comment]
[xanderdunn](https://github.com/xanderdunn) - thx for confirming the performance gap was addressed (and improved!)
